### ISSUE TITLE
Remove HTTP server from emulation core

### ISSUE THREAD

**Issue by ajrhacker** [mamedev/mame#9055]
To me, the HTTP server does not seem like something that really belongs in the `src/emu` core, much less included within `emu.h` as it is now. It looks like the kind of feature that might be possible to implement as a Lua plugin nowadays; if that approach is not workable, it should be managed as a frontend resource.

Presently, one device implementation (src/mame/machine/esqpanel.cpp) does depend specifically on the HTTP server interface; this part would have to be rewritten or eliminated.

**Comment by DavidHaywood** [comment]
for systems where the software was 100% digital, or where an update service was required that we need to simulate, I can imagine the need to internal HTTP, FTP and maybe even custom protocols to simulate servers which no longer exist and still allow the software to function.

I don't know if this is what this implementation was for, but such a case probably would want to use code in the project, not scripts.

**Comment by smf-** [comment]
This doesn't appear to be for that, if you run

mame esq1 -http

then point a browser at

http://localhost:8080/esqpanel/vfx/FrontPanel.html

Then you'll get a black screen.

I'm not 100% sure what it's supposed to actually do, I can't find
anything using google or on the wiki.

**Comment by Firehawke** [comment]
It used to-- a long time ago-- act as an additional UI interface. There were two problems that let it bitrot.

One is that the license on the web server code we were using changed, making it impossible for us to continue to use it. It was then ripped out. Someone added an alternative one down the road, but it was never fully implemented and the second problem ties in a lot with that:

Two is that getting anything at all to interface to MAME's UI code is a nightmare. The bits I'd had implemented and working back in 2013 were fairly shaky. I've stated on multiple occasions that if a proper, documented set of Lua hooks were in place to handle everything, I'd revisit the old project and see if I couldn't resurrect it.

As far as I know, the actual task of untangling the UI is a Sispyphean nightmare and probably won't ever happen.

**Comment by cracyc** [comment]
@npwoods uses lua hooks to control mame.  I'm sure they could be used for an http interface.

**Comment by simzy39** [comment]
Someone must use it since there is the pull #3716 https://github.com/mamedev/mame/pull/3716

**Comment by cuavas** [comment]
Yeah, I think the HTTP server is an overly specific thing to have in src/emu.  The way the Ensoniq driver hooks into it now is overly coupled, too.  I’d agree that the web server needs better integration with the Lua features, so scripts can provide glue to give a web interface to specific devices or drivers.

**Comment by cbrunschen** [comment]
The keyboards that offer a web-based panel are the VFX family, so `vfx`, `vfxsd`, `sd1` and `sd132`. So of you run

`mame vfx -http`

(or `vfxsd`, `sd1` or `sd132`)

and connect to

http://localhost:8080/esqpanel/FrontPanel.html

you should see a working front panel for one of the VFX family of keyboards.

Of interest may be https://github.com/mamedev/mame/pull/14131 which 

- adds internal layout-based panels for the VFX family of keyboards
- entirely _removes_ their use of the HTTP server
  - instead adds a small asio-based server that exposes a simple text-based protocol for external panel software to use

and which thus would make removal of the HTTP server easier.
